### PR TITLE
(feat): Adding control plane validation experiment in pipeline

### PR DESCRIPTION
**Pull request by rajdas98**
Signed-off-by: Raj <raj.das@mayadata.io>
- Adding control plane validation experiment in pipeline

### CODE REVIEW

File: tests/openebs-control-plane-validation_test.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+	// BDD TEST CASE 1
+	Context("Check for the components of openebs", func() {
+
+		It("Should check for creation of runner pod", func() {
+
+			
+			//Creating rbac for the experiment
+			By("Creating rbac for the experiment")
+			err = exec.Command("kubectl", "apply", "-f", "https://raw.githubusercontent.com/litmuschaos/chaos-charts/master/charts/openebs/openebs-control-plane-validation/rbac.yaml", "-n", "openebs").Run()
+			Expect(err).To(BeNil(), "fail to create chaos experiment")
```
Comment (uditgaurav) [inline tests/openebs-control-plane-validation_test.go]: fail to create rbac for the experiment

File: tests/openebs-control-plane-validation_test.go [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+			if err != nil {
+				fmt.Println(err)
+			}
```
Comment (uditgaurav) [inline tests/openebs-control-plane-validation_test.go]: if error is not nil the code will fail on above line only .. we don't need this `if-condition`

File: tests/openebs-control-plane-validation_test.go [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+			//Creating Chaos-Experiment
+			By("Creating Experiment")
+			err = exec.Command("kubectl", "apply", "-f", "https://raw.githubusercontent.com/litmuschaos/chaos-charts/master/charts/openebs/openebs-control-plane-validation/rbac.yaml", "-n", "openebs").Run()
```
Comment (uditgaurav) [inline tests/openebs-control-plane-validation_test.go]: Why are we creating rbac again? we should apply the experiment link here?

File: tests/openebs-control-plane-validation_test.go [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+			Expect(err).To(BeNil(), "fail to create chaos experiment")
+			if err != nil {
+				fmt.Println(err)
+			}
+
```
Comment (uditgaurav) [inline tests/openebs-control-plane-validation_test.go]: same here we can remove `if-condition`

File: tests/openebs-control-plane-validation_test.go (hunk 2 above)
Comment (uditgaurav) [reply tests/openebs-control-plane-validation_test.go]: please remove in all places where we have written like this.

File: tests/openebs-control-plane-validation_test.go (hunk 3 above)
Comment (uditgaurav) [reply tests/openebs-control-plane-validation_test.go]: use hub link for the experiment installation.

File: tests/openebs-control-plane-validation_test.go [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+			fmt.Println("Chaosengine created successfully...")
+
+			//Fetching engine-nginx-runner pod
+			runner, err := client.CoreV1().Pods("openebs").Get("control-plane-chaos-runner", metav1.GetOptions{})
+			fmt.Printf("name : %v \n", runner.Name)
+			for i := 0; i < 30; i++ {
+				if string(runner.Status.Phase) != "Succeeded" {
+					time.Sleep(10 * time.Second)
+					runner, _ = client.CoreV1().Pods("openebs").Get("control-plane-chaos-runner", metav1.GetOptions{})
+					fmt.Printf("Current Runner is in %v State, Please Wait ...\n", runner.Status.Phase)
```
Comment (uditgaurav) [inline tests/openebs-control-plane-validation_test.go]: `fmt.Printf("Currently the  Runner pod is in %v State, Please Wait ...\n", runner.Status.Phase)`

File: tests/openebs-control-plane-validation_test.go [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+				} else {
+					break
+				}
+			}
+			Expect(err).To(BeNil())
```
Comment (uditgaurav) [inline tests/openebs-control-plane-validation_test.go]: give an error message also

File: tests/openebs-control-plane-validation_test.go [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+			Expect(string(runner.Status.Phase)).To(Or(Equal("Running"), Equal("Succeeded")))
+
+			//Checking the chaosresult
+			By("Checking the chaosresult")
+			app, _ := clientSet.ChaosResults("openebs").Get("control-plane-chaos-openebs-control-plane-validation", metav1.GetOptions{})
+			Expect(string(app.Spec.ExperimentStatus.Verdict)).To(Equal("Pass"), "Verdict is not pass in chaosresult")
```
Comment (uditgaurav) [inline tests/openebs-control-plane-validation_test.go]: Is the value inside Get function is correct?

Comment (rajdas98) [reply tests/openebs-control-plane-validation_test.go]:
```=== RUN   TestChaos
Running Suite: BDD test
=======================
Random Seed: 1582659291
Will run 1 of 1 specs

Chaosengine created successfully...
name : control-plane-chaos-runner 
Currently the Runner pod is in Running State, Please Wait ...
Currently the Runner pod is in Running State, Please Wait ...
Currently the Runner pod is in Running State, Please Wait ...
Currently the Runner pod is in Running State, Please Wait ...
Currently the Runner pod is in Running State, Please Wait ...
Currently the Runner pod is in Running State, Please Wait ...
Currently the Runner pod is in Running State, Please Wait ...
Currently the Runner pod is in Running State, Please Wait ...
Currently the Runner pod is in Running State, Please Wait ...
Currently the Runner pod is in Running State, Please Wait ...
Currently the Runner pod is in Running State, Please Wait ...
Currently the Runner pod is in Running State, Please Wait ...
Currently the Runner pod is in Running State, Please Wait ...
Currently the Runner pod is in Running State, Please Wait ...
Currently the Runner pod is in Running State, Please Wait ...
Currently the Runner pod is in Running State, Please Wait ...
Currently the Runner pod is in Succeeded State, Please Wait ...
• [SLOW TEST:178.404 seconds]
BDD of openebs control plane chaos experiment
/home/rajdas/go/src/github.com/rajdas98/litmus-e2e/tests/openebs-control-plane-validation_test.go:66
  Check for the components of openebs
  /home/rajdas/go/src/github.com/rajdas98/litmus-e2e/tests/openebs-control-plane-validation_test.go:69
    Should check for creation of runner pod
    /home/rajdas/go/src/github.com/rajdas98/litmus-e2e/tests/openebs-control-plane-validation_test.go:71
------------------------------

Ran 1 of 1 Specs in 178.407 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestChaos (178.41s)
PASS
ok  	command-line-arguments	178.438s
```

**Comment by rajdas98** [comment]
```=== RUN   TestChaos
Running Suite: BDD test
=======================
Random Seed: 1582659291
Will run 1 of 1 specs

Chaosengine created successfully...
name : control-plane-chaos-runner 
Currently the Runner pod is in Running State, Please Wait ...
Currently the Runner pod is in Running State, Please Wait ...
Currently the Runner pod is in Running State, Please Wait ...
Currently the Runner pod is in Running State, Please Wait ...
Currently the Runner pod is in Running State, Please Wait ...
Currently the Runner pod is in Running State, Please Wait ...
Currently the Runner pod is in Running State, Please Wait ...
Currently the Runner pod is in Running State, Please Wait ...
Currently the Runner pod is in Running State, Please Wait ...
Currently the Runner pod is in Running State, Please Wait ...
Currently the Runner pod is in Running State, Please Wait ...
Currently the Runner pod is in Running State, Please Wait ...
Currently the Runner pod is in Running State, Please Wait ...
Currently the Runner pod is in Running State, Please Wait ...
Currently the Runner pod is in Running State, Please Wait ...
Currently the Runner pod is in Running State, Please Wait ...
Currently the Runner pod is in Succeeded State, Please Wait ...
• [SLOW TEST:178.404 seconds]
BDD of openebs control plane chaos experiment
/home/rajdas/go/src/github.com/rajdas98/litmus-e2e/tests/openebs-control-plane-validation_test.go:66
  Check for the components of openebs
  /home/rajdas/go/src/github.com/rajdas98/litmus-e2e/tests/openebs-control-plane-validation_test.go:69
    Should check for creation of runner pod
    /home/rajdas/go/src/github.com/rajdas98/litmus-e2e/tests/openebs-control-plane-validation_test.go:71
------------------------------

Ran 1 of 1 Specs in 178.407 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestChaos (178.41s)
PASS
ok  	command-line-arguments	178.438s
```

File: tests/openebs-control-plane-validation_test.go (hunk 7 above)
Comment (rajdas98) [reply tests/openebs-control-plane-validation_test.go]: Yes it is correct

File: tests/openebs-control-plane-validation_test.go (hunk 6 above)
Comment (rajdas98) [reply tests/openebs-control-plane-validation_test.go]: Done

File: tests/openebs-control-plane-validation_test.go (hunk 5 above)
Comment (rajdas98) [reply tests/openebs-control-plane-validation_test.go]: Done

File: tests/openebs-control-plane-validation_test.go (hunk 3 above)
Comment (rajdas98) [reply tests/openebs-control-plane-validation_test.go]: Done

File: tests/openebs-control-plane-validation_test.go (hunk 1 above)
Comment (rajdas98) [reply tests/openebs-control-plane-validation_test.go]: Done

**Pull request merged**